### ISSUE TITLE
Site editor: Editor canvas iframe focus style doesn't work with Safari

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51496]
### Description

In teh Site Editor, when navigating with the Tab key and focus lands on the editor canvas iframe, a blue outline focus style is drawn around the iframe. Screenshot:

<img width="1792" alt="Screenshot 2023-06-14 at 16 05 22" src="https://github.com/WordPress/gutenberg/assets/1682452/6e9dcacd-7329-4fce-a08e-58c86e702a85">

This focus style doesn't work with Safari. It appears that in Safari, outlines with a negative offset set on an iframe element are not visible, depending on some other CSS properties. See [a reduced test case on this codepen](https://codepen.io/afercia/full/jOQbjJw), and test with Chrome, Firefox, and Safari.

I spent a good amount of time in debugging but I wasn't able to determine the root cause of this Safari buggy behavior. Seems to me it's a combination of some CSS properties on differemt elements:
- background color on the iframe boxy element
- margin value on the iframe body element
- border value on the iframe element

I'd defer to someone else with more advanced CSS knowledge but I'd suggest to consider to not use the outline property and try an alternative method instead. Cc @jasmussen 

### Step-by-step reproduction instructions

- Tesst with Chrome or Firefox.
- Go to the Site editor, stay in the initial browse mode.
- Use the Tab key to navigate the UI.
- Once focus lands on the editor canvas iframe, observe there's a visible blue outline around the iframe.
- Now test with Safari.
- Repeat the steps above.
- Observe there's no visible blue outline.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jasmussen** [comment]
Can reproduce, yes, this is a tricky one. Border works, but then that pushes the content inwards. Box shadow doesn't seem to work, and pseudo elements don't either. I'd appreciate a broader CSS insight from @WordPress/gutenberg-design

**Comment by jameskoster** [comment]
> Border works, but then that pushes the content inwards

It's not ideal, but we can fix the 'pushing' issue by supplying a transparent border on the non-focussed iframe. 

Beyond that I'm struggling a bit for ideas... we might need an unorthodox focus style for this element?

**Comment by jasmussen** [comment]
> It's not ideal, but we can fix the 'pushing' issue by supplying a transparent border on the non-focussed iframe.

We would need to compensate for this border both in the margin and the border radius. But I guess the math can be worked out in SCSS.

**Comment by jameskoster** [comment]
That wouldn't be the end of the world, but nice to avoid if possible.

Is there another way we could indicate focus? Increase the radius, or the scale (IE a more pronounced version of the hover effect)?
